src/task/create-item-validation-task.ts

```diff
@@ -0,0 +1,142 @@
+import { ItemValidationProcess } from '../types';
+
+type InputType = { itemId: string };
+
+export class CreateItemValidationTask extends BaseValidationTask<string> {
+  input: InputType;
+  itemService: ItemService;
+  fTM: FileTaskManager;
+  runner: TaskRunner<Actor>;
+  serviceItemType: string;
+
+  get name(): string {
+    return CreateItemValidationTask.name;
+  }
+
+  constructor(
+    member: Member,
+    validationService: ItemValidationService,
+    itemService: ItemService,
+    fTM: FileTaskManager,
+    runner: TaskRunner<Actor>,
+    serviceItemType: string,
+    input: InputType,
+  ) {
+    super(member, validationService);
+    this.itemService = itemService;
+    this.input = input;
+    this.fTM = fTM;
+    this.runner = runner;
+    this.serviceItemType = serviceItemType;
+  }
+
+  async run(handler: DatabaseTransactionHandler): Promise<void> {
+    this.status = 'RUNNING';
+
+    const { itemId } = this.input;
+
+    // create record in item-validation
+    const iVId = await this.validationService.createItemValidation(
+      itemId,
+      handler,
+    );
+    // variable to indicate if manual review is needed
+    let needReview = false;
+
+    // get all enabled processes
+    const enabledProcesses = await this.validationService.getEnabledProcesses(handler);
+
+    // get map for all statuses
+    const iVStatuses = await this.validationService.getItemValidationStatuses(handler);
+
+    // get item
+    const item = await this.itemService.get(itemId, handler);
+
+    // recursively validate item and subitems
+    const validateItem = async (item: Item) => {
+
+      const executeProcess = async (process: ItemValidationProcess) => {
+        if (process?.name === ItemValidationProcesses.ImageChecking && item?.type !== this.serviceItemType)
+        return;
+
+        // create pending entry
+        const iVGEntry = await this.validationService.createItemValidationGroup(
+          item?.id,
+          iVId,
+          process.id,
+          getStatusIdByName(iVStatuses, ItemValidationStatuses.Pending),
+          handler,
+        ).then(data => data);
+
+        const status = await handleProcesses(process, item, this.fTM, this.actor, this.runner).catch(error => {
+          // log the error
+          console.error(error);
+          // if some error happend during the execution of a process, we consider the it failed
+          return ItemValidationStatuses.Failure;
+        });
+        
+        if (status === '')
+          throw new ProcessNotFoundError(process.name);
+
+        // need review if any process failed
+        if (status === ItemValidationStatuses.Failure) needReview = true;
+
+        console.log('prepare to update: ', iVGEntry.id);
+        // update status
+        const updatedEntry = await this.validationService.updateItemValidationGroup(iVGEntry.id, getStatusIdByName(iVStatuses, status), handler);
+        console.log('item validation group entry updated: ', updatedEntry);
```

Comment:
Actually `run` provides a `logger` or `log` it would be best to use it instead of console.log.